src/services/code-index/managed/scanner.ts

```diff
@@ -0,0 +1,428 @@
+
+		// Determine which files to scan
+		const filesToScan = await getFilesToScan(config.workspacePath, currentBranch, isBase)
+
+		console.info(`Scanning ${filesToScan.length} files on branch ${currentBranch} (isBase: ${isBase})`)
+
+		// Process files with manifest for intelligent skipping
+		const result = await processFiles(filesToScan, config, context, currentBranch, isBase, manifest, onProgress)
+
+		return {
+			success: result.errors.length === 0,
+			filesProcessed: result.filesProcessed,
+			filesSkipped: result.filesSkipped,
+			chunksIndexed: result.chunksIndexed,
+			errors: result.errors,
+		}
+	} catch (error) {
+		const err = error instanceof Error ? error : new Error(String(error))
+		errors.push(err)
+		console.error(`Scan directory failed: ${err.message}`)
+		TelemetryService.instance.captureEvent(TelemetryEventName.CODE_INDEX_ERROR, {
+			error: err.message,
+			stack: err.stack,
+			location: "scanDirectory",
+		})
+
+		return {
+			success: false,
+			filesProcessed: 0,
+			filesSkipped: 0,
+			chunksIndexed: 0,
+			errors,
+		}
+	}
+}
+
+/**
+ * Determines which files to scan based on branch strategy
+ *
+ * @param workspacePath Workspace root path
+ * @param currentBranch Current git branch
+ * @param isBase Whether current branch is a base branch
+ * @returns Array of file paths to scan
+ */
+async function getFilesToScan(workspacePath: string, currentBranch: string, isBase: boolean): Promise<string[]> {
+	if (isBase) {
+		// Base branch: scan all files
+		return await getAllSupportedFiles(workspacePath)
+	} else {
+		// Feature branch: only scan changed files
+		const baseBranch = await getBaseBranch(workspacePath)
+		const diff = await getGitDiff(currentBranch, baseBranch, workspacePath)
+		const changedFiles = [...diff.added, ...diff.modified]
+
+		// Convert relative paths from git to absolute paths and filter to only supported files
+		return changedFiles
+			.filter((file) => {
+				const ext = path.extname(file).toLowerCase()
+				return scannerExtensions.includes(ext)
+			})
+			.map((file) => path.join(workspacePath, file))
+	}
+}
+
+/**
+ * Gets all supported files in the workspace that are tracked by git
+ *
+ * @param workspacePath Workspace root path
+ * @returns Array of supported file paths (absolute paths)
+ */
+async function getAllSupportedFiles(workspacePath: string): Promise<string[]> {
+	// Get all git-tracked files (relative paths) using async generator
+	const gitTrackedFiles: string[] = []
+	for await (const file of getGitTrackedFiles(workspacePath)) {
+		gitTrackedFiles.push(file)
+	}
+
+	logger.info(`Found ${gitTrackedFiles.length} git-tracked files`)
+
+	// Initialize ignore controller for .rooignore
+	const ignoreController = new RooIgnoreController(workspacePath)
+	await ignoreController.initialize()
+
+	// Filter by .rooignore
+	const allowedPaths = ignoreController.filterPaths(gitTrackedFiles)
+
+	logger.info(`After .rooignore filter: ${allowedPaths.length} files`)
```

Comment:
Nice– I was hoping you were already using the IgnoreController for this code and you are! :)